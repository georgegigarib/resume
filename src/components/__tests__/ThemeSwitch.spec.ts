import { mount, VueWrapper } from '@vue/test-utils'
import { describe, it, expect, vi, afterEach, beforeEach } from 'vitest'
import ThemeSwitch from '@/components/ThemeSwitchTwo.vue'
import { nextTick } from 'vue'

// Mocking the VueUse functions
vi.mock('@vueuse/core', () => ({
  useDark: vi.fn(() => ({
    value: false
  })),
  useToggle: vi.fn(() => vi.fn())
}))

describe('ThemeSwitch', () => {
  let wrapper: VueWrapper<any>

  beforeEach(() => {
    wrapper = mount(ThemeSwitch)
  })

  it('should mount the component', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should switch between light and dark themes', async () => {
    const checkbox = wrapper.find('input[type="checkbox"]')

    expect((checkbox.element as HTMLInputElement).checked).toBe(false)

    // Simulate checkbox click
    await checkbox.setValue(true)
    await checkbox.trigger('change')
    await nextTick()

    expect((checkbox.element as HTMLInputElement).checked).toBe(true)
    expect(wrapper.vm.switchValue).toBe(true)
  })

  it('should accept size prop and apply it', async () => {
    const size = 20
    await wrapper.setProps({ size })
    expect(wrapper.vm.size).toBe(size)
  })

  it('should have default size when size prop is not provided', () => {
    expect(wrapper.vm.size).toBe(30)
  })

  afterEach(() => {
    vi.clearAllMocks()
    if (wrapper) {
      wrapper.unmount()
    }
  })
})
