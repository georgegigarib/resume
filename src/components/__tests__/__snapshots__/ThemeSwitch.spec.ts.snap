// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeSwitch > should mount the component 1`] = `"<label data-v-92108eaa="" class="switch"><input data-v-92108eaa="" type="checkbox"><span data-v-92108eaa="" class="slider"></span></label>"`;
